lib/api/apiUtils/authorization/permissionChecks.js

```diff
@@ -346,6 +346,15 @@ function isBucketAuthorized(bucket, requestTypesInput, canonicalID, authInfo, lo
             return results[_requestType];
         }
         const aclPermission = checkBucketAcls(bucket, _requestType, canonicalID, mainApiCall);
+        // In case of error bucket access is checked with bucketGet
+        // For website, bucket policy only uses objectGet and ignores bucketGet
+        // https://docs.aws.amazon.com/AmazonS3/latest/userguide/WebsiteAccessPermissionsReqd.html
+        // bucketGet should be used to check acl but switched to objectGet for bucket policy
+        if (isWebsite && _requestType === 'bucketGet') {
+            // eslint-disable-next-line no-param-reassign
+            _requestType = 'objectGet';
+            actionImplicitDenies[_requestType] = actionImplicitDenies[_requestType] || false;
```

Comment:
minor clarity suggestion

```suggestion
            actionImplicitDenies['objectGet'] = actionImplicitDenies['objectGet'] || false;
```